### PR TITLE
Move collecting and storing multiple pings at once into Rust core

### PR DESCRIPTION
**Pull request by badboy**
This should enable storing pings concurrently eventually.

This is in part to address [bug 1553813](https://bugzilla.mozilla.org/show_bug.cgi?id=1553813), though it does nothing in parallel yet.
We don't know if that's an actuall bottleneck and right now I think we would just run into lock contention on the database anyway (plus right now we send _at most_ 2 pings at once)

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt [hunk 1]
```diff
@@ -450,16 +437,17 @@ open class GleanInternalAPI internal constructor () {
      * @return The async [Job] performing the work of assembling the ping
      */
     internal fun sendPings(pings: List<PingType>): Job? {
-        val sendPing: (PingType) -> Boolean = {
-            LibGleanFFI.INSTANCE.glean_send_ping(
+        val pingArray = pings.map { it.handle }.toLongArray()
```
[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: We could indeed do that. We then need to _copy_ the strings into the string array though, send that over FFI and reconstruct the strings by _copying them onto the heap_

[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: Given how low frequency this operation should be, maybe we shouldn't bother too much and use `sendPingByName`. @mdboom thoughts?

[the remark this answers is not in the document]
Comment (mdboom) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: I doubt performance matters at this level -- it's probably dominated by "collecting" the pings if anything -- and this PR at least makes optimizing that part possible down the road by moving it into Rust.

Always using `sendPingByName` would of course also have the overhead of looking up the ping in the name-to-ping HashMap (again, that map is so small, it's doubtful even measurable).

So :man_shrugging:.  Whatever -- all these options seem to move things forward.

[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: `sendPingByName`! Go for it @badboy !

[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: I'm disappointed. You could have done `sendPingByName("badboy")`.

[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt]: I can only distribute so much happiness and humour in a single day! :P

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt [hunk 2]
```diff
@@ -389,6 +389,11 @@ class GleanTest {
 
         val server = getMockWebServer()
 
+        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
```
Comment (Dexterp37) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: Why is this needed? Is it because of glean handles?

Comment (badboy) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: Yes, pings are registered through the FFI, but if we reset Glean we also clear the object on the Rust side, at which point we don't have the info around to re-register pings. (technically we _could_ have the data around if we put it into the `pingTypeQueue`, but when I tried that the simple way I ran into multi-threading problems)
